tests/BanditClientTest.php

```diff
@@ -0,0 +1,272 @@
+use Eppo\DTO\Bandit\Bandit;
+use Eppo\DTO\Bandit\BanditEvaluation;
+use Eppo\DTO\Bandit\BanditModelData;
+use Eppo\DTO\Bandit\BanditResult;
+use Eppo\EppoClient;
+use Eppo\Exception\EppoClientException;
+use Eppo\Exception\EppoException;
+use Eppo\Logger\BanditActionEvent;
+use Eppo\Logger\LoggerInterface;
+use Eppo\PollerInterface;
+use Eppo\Tests\WebServer\MockWebServer;
+use Exception;
+use PHPUnit\Framework\TestCase;
+
+class BanditClientTest extends TestCase
+{
+    private const EXPERIMENT_NAME = 'numeric_flag';
+    private const TEST_DATA_PATH = __DIR__ . '/data/ufc/bandit-tests';
+
+    public static function setUpBeforeClass(): void
+    {
+        try {
+            MockWebServer::start(__DIR__ . '/data/ufc/bandit-flags-v1.json');
+        } catch (Exception $exception) {
+            self::fail('Failed to start mocked web server: ' . $exception->getMessage());
+        }
+
+        try {
+            EppoClient::init('dummy', 'http://localhost:4000',);
+        } catch (Exception $exception) {
+            self::fail('Failed to initialize EppoClient: ' . $exception->getMessage());
+        }
+    }
+
+    public static function tearDownAfterClass(): void
+    {
+        MockWebServer::stop();
+        DefaultCacheFactory::clearCache();
+    }
+
+    public function testBanditWithEmptyActions(): void
+    {
+        $flagKey = 'bandit';
+        $actions = [];
+        $subjectKey = 'user123';
+        $subject = ['country' => 'USA', 'age' => 25];
+        $default = 'defaultVariation';
+
+
+        $config = $this->getMockBuilder(ConfigurationLoader::class)->disableOriginalConstructor()->getMock();
+
+        $config->expects($this->once())
+            ->method('isBanditFlag')
+            ->with($flagKey)
+            ->willReturn(true);
+
+        $client = EppoClient::createTestClient($config, poller: $this->getPollerMock());
+
+        $this->expectException(EppoClientException::class);
+        $this->expectExceptionCode(EppoException::BANDIT_EVALUATION_FAILED_NO_ACTIONS_PROVIDED);
+
+        $client->getBanditAction($flagKey, $subjectKey, $subject, $actions, $default);
+    }
+
+    public function testNonBandit(): void
+    {
+        $flagKey = 'non_bandit';
+        $actions = [];
+        $subjectKey = 'user123';
+        $subject = ['country' => 'USA', 'age' => 25];
+        $default = 'defaultVariation';
+
+
+        $config = $this->getMockBuilder(ConfigurationLoader::class)->disableOriginalConstructor()->getMock();
+
+        $config->expects($this->once())
+            ->method('isBanditFlag')
+            ->with($flagKey)
+            ->willReturn(false);
+
+
+        $client = EppoClient::createTestClient($config, poller: $this->getPollerMock());
+
+
+        $result = $client->getBanditAction($flagKey, $subjectKey, $subject, $actions, $default);
+        $this->assertEquals($default, $result->variation);
+        $this->assertEquals(null, $result->action);
+    }
+
+    public function testBanditModelDNE(): void
```

Comment:
Consider spelling out `DNE` as I didn't realize what that meant until I read the test

Reply:
done